crates/bgp-pkt/src/wire/tests/bgp_ls.rs

```diff
@@ -0,0 +1,398 @@
+use ipnet::{IpNet, Ipv4Net, Ipv6Net};
+use netgauze_parse_utils::WritablePdu;
+
+#[test]
+fn test_wire() -> Result<(), PathAttributeWritingError> {
+    let good_wire = [
+        0x90, 0x0e, 0x00, 0x6d, 0x40, 0x04, 0x47, 0x10, 0x20, 0x01, 0x0d, 0xb8, 0x00, 0x01, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x01, 0x00, 0x28, 0x01,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x1b, 0x02, 0x00, 0x00,
+        0x04, 0x00, 0x01, 0x00, 0x00, 0x02, 0x01, 0x00, 0x04, 0x00, 0x00, 0x00, 0x00, 0x02, 0x03,
+        0x00, 0x07, 0x00, 0x00, 0x00, 0x00, 0x00, 0x09, 0x01, 0x00, 0x01, 0x00, 0x28, 0x01, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x1b, 0x02, 0x00, 0x00, 0x04,
+        0x00, 0x01, 0x00, 0x00, 0x02, 0x01, 0x00, 0x04, 0x00, 0x00, 0x00, 0x00, 0x02, 0x03, 0x00,
+        0x07, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x03,
+    ];
+
+    let good = PathAttribute::from(
+        true,
+        false,
+        false,
+        true,
+        PathAttributeValue::MpReach(MpReach::BgpLs {
+            next_hop: IpAddr::V6(Ipv6Addr::from_str("2001:db8:1::1").unwrap()),
+            nlri: vec![
+                BgpLsNlri {
+                    path_id: None,
+                    value: BgpLsNlriValue::Node(BgpLsNlriNode {
+                        protocol_id: BgpLsProtocolId::IsIsLevel1,
+                        identifier: 0,
+                        local_node_descriptors: BgpLsNodeDescriptorTlv::Local(vec![
+                            BgpLsNodeDescriptorSubTlv::AutonomousSystem(65536),
+                            BgpLsNodeDescriptorSubTlv::BgpLsIdentifier(0),
+                            BgpLsNodeDescriptorSubTlv::IgpRouterId(vec![0, 0, 0, 0, 0, 9, 1]),
+                        ]),
+                    }),
+                },
+                BgpLsNlri {
+                    path_id: None,
+                    value: BgpLsNlriValue::Node(BgpLsNlriNode {
+                        protocol_id: BgpLsProtocolId::IsIsLevel1,
+                        identifier: 0,
+                        local_node_descriptors: BgpLsNodeDescriptorTlv::Local(vec![
+                            BgpLsNodeDescriptorSubTlv::AutonomousSystem(65536),
+                            BgpLsNodeDescriptorSubTlv::BgpLsIdentifier(0),
+                            BgpLsNodeDescriptorSubTlv::IgpRouterId(vec![0, 0, 0, 0, 0, 1, 3]),
+                        ]),
+                    }),
+                },
+            ],
+        }),
+    )
+    .unwrap();
+
+    test_parsed_completely_with_one_input(
+        &good_wire,
+        &mut BgpParsingContext::new(
+            true,
+            HashMap::new(),
+            HashMap::new(),
+            false,
+            false,
+            false,
+            false,
+        ),
+        &good,
+    );
+    test_write(&good, &good_wire)?;
+
+    Ok(())
+}
+
+#[test]
+pub fn test_bgp_ls_attr_parse() {
+    let value = BgpLsAttribute {
```

Comment:
I would follow the pattern used in other tests: see https://github.com/NetGauze/NetGauze/blob/main/crates/bgp-pkt/src/wire/tests/path_attribute.rs#L352

Basically: define what's the expected wire byte array, the expected struct, then use one of `test_parsed_completely_*` variants to test that the byte array parsing will result the same value. After use `test_write` to test the serialization part.